### ISSUE TITLE
model saving

### ISSUE THREAD

**Issue by ridmajayasundara** [cepdnaclk/e18-4yp-GPU-Acceleration-for-Deep-Learning-based-Comprehensive-ECG-analysis#33]
**as of now models are saved as the complete model.
instead of that way, we can consider saving the model weights.**

Initialize the minimum validation loss and the best model state
`min_val_loss = float('inf')`
`best_model_state_dict = None`

Calculate the average validation loss
`    avg_val_loss = val_loss / (len(val_dataloader) * batch_size)`

Save the model state if the validation loss is lower than the current minimum
`    if avg_val_loss < min_val_loss:`
`        min_val_loss = avg_val_loss`
`        best_model_state_dict = model.state_dict()`

Load the best model state and save it
`model.load_state_dict(best_model_state_dict)`

Save the best model with date and time in the path
`current_time = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")`
`model_path = f"saved_models/{current_time}_best_model"`
`torch.save(model.state_dict(), model_path)`
